src/test/client.spec.ts

```diff
@@ -158,6 +158,32 @@ describe('Client', () => {
       expect(task.interval).toBe(60000);
     });
 
+    it('supports ending a task at a user defined end date', async () => {
+      const startTime = moment()
+        .add(5, 'minutes')
+        .toDate();
+
+      const endTime = moment()
+        .add(4, 'minutes')
+        .toDate();
+
+      const task = await client.create(
+        type,
+        { hello: 'world' },
+        {
+          interval: 60000, // 10 minutes
+          scheduledTime: startTime,
+          taskEndTime: endTime
+        }
+      );
+
+      // The task will fail because it will have no last run and no next run
+      expect(task.status).toBe(TaskStatus.Failed);
```

Comment:
i may be forgetting something, but why would this be failed? it never ran, right?

Reply:
So I checked the logic and it goes something like this. If the nextRunTime is undefined and there is no last run then status is failed. There is a test that checks that - https://github.com/princjef/irontask/blob/af4a22c771df99a14354a423dff6ff2f74cc8569/src/task/status.spec.ts#L235

Reply:
cool, glad that was an explicit case :) I'm assuming this is acceptable behavior for your use case?

Reply:
Yup that is acceptable